authors/apps/articles/views.py

```diff
@@ -84,4 +99,81 @@ def destroy(self, request, pk=None):
         article = get_object_or_404(queryset, pk=pk)
         article.delete()
         return Response({"message": "article deleted successfully"},
-                        status=status.HTTP_200_OK)
+                        status=status.HTTP_204_NO_CONTENT)
+
+
+class RatingAPIView(GenericAPIView):
+    queryset = Rating.objects.all()
+    serializer_class = RatingSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    # renderer_classes = (RatingJsonRenderer,)
+
+    def post(self, request, slug):
+        """POST request to rate an article."""
+        rating = request.data
+        article = get_article(slug)
+
+        if not article:
+            raise ValidationError(
+                detail={'message': 'Article not found'}
+            )
+
+        if request.user.id == article.author.id:
+            return Response({
+                "message": "You cannot rate your own article"
+            }, status=status.HTTP_403_FORBIDDEN)
+
+        try:
+            current_rating = Rating.objects.get(
+                user=request.user.id,
+                article=article
+            )
+            serializer = self.serializer_class(
+                current_rating, data=rating)
+        except Rating.DoesNotExist:
+            serializer = self.serializer_class(data=rating)
+
+        serializer.is_valid(raise_exception=True)
+        serializer.save(user=request.user, article=article)
+
+        return Response({
+            'message': 'Rating submitted sucessfully',
+            'data': serializer.data
+        }, status=status.HTTP_201_CREATED)
+
+    def get(self, request, slug):
+        """Get request for an article ratings."""
+        rating = None
+        try:
```

Comment:
Can't you use the method defined in line 15 as it was used in line 114?

Reply:
When I use it I get the error I informed you about